fluent/src/builtins.js

```diff
@@ -31,3 +24,25 @@ function values(opts) {
   }
   return unwrapped;
 }
+
+export
+function NUMBER([arg], opts) {
+  if (arg instanceof FluentNone) {
+    return arg;
+  }
+  if (arg instanceof FluentNumber) {
+    return new FluentNumber(arg.valueOf(), merge(arg.opts, opts));
+  }
+  return new FluentNone("NUMBER()");
```

Comment:
What's the intention for returning `NUMBER()` as the fallback string? I'm concerned that it locks down how we display errors based on something slightly misleading - if I have `key = Foo { NUMBER($emailCount, minimumFractionDigits: 4) }` then displaying `Foo [NUMBER()]` may be misleading.

On top of that, from the design perspective, in `fluent-rs` I leaned on the side of keeping display formatting located in one place, rather than each function returning its own representation.

So, to translate it to JS, I'd expect this to return sth like `FluentError({type: "function", name: "NUMBER"})` and optionally `args` so that some `toString` of `FluentError` can construct the visual representation, for example `NUMBER()` like here, or something else.

The added benefit is that since you're passing semantic information rather than stringified representation, you can differentiate how you show the `FluentError` between what you send to console, and what you display in the formatter partial string.

It's not a blocker, just a suggestion.

Reply:
> What's the intention for returning NUMBER() as the fallback string?

This happens when the arg resolves cleanly, but it's not the right type, e.g. `NUMBER($userName)`. Right now this prints `NaN` which I'd almost consider a bug. In this PR, I'm changing it to `NUMBER()`. I think ideally the `$userName` reference would store both the resolved value of the variable _and_ its name, so that when the `NUMBER` functions fails, the output can use the name for fallback: `$userName` or `NUMBER($userName)`.

I think this again would benefit from the resolver spec, so in this PR, I'd just like to fix the `NaN` thing.

Reply:
I think that in the long run, we'll add more information to `FluentType` instances, including the detailed error information in `FluentNone`. The `fluent-rs` examples look like something we could base the design on.